transport/http/handler.go

```diff
@@ -144,8 +143,12 @@ func handleOnewayRequest(
 	}
 	treq.Body = &buff
 
+	// create a new context for oneway requests since the HTTP handler cancels
+	// http.Request's context when ServeHTTP returns
+	ctx := opentracing.ContextWithSpan(context.Background(), span)
```

Comment:
Just read your second comment kris.

There's definitely a spectrum of what guarantees a oneway transport offers: from 'try once' to  're-process until success'. HTTP will likely be the most lossey oneway transport we have (akin to UDP as you mentioned).

However, I'm not sure of the best way to make this clear to users.